benchmarks/quant-hela-method-comparison/scripts/01_download_data.py

```
@@ -0,0 +1,259 @@
+        urllib.request.urlopen(request, timeout=10)
+        return True
+    except:
+        return False
+
+
+def list_available_files(base_url: str) -> List[str]:
+    """
+    List files available at a URL (works for directory listings).
+    """
+    try:
+        response = urllib.request.urlopen(base_url, timeout=30)
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)